src/site/content/en/blog/why-coop-coep/index.md

```diff
@@ -0,0 +1,241 @@
+attacks a web page could carry out. We call it a cross-origin isolated state.
+This is exactly what COOP+COEP is about.
+
+Under a cross-origin isolated state, the requesting site is considered less
+dangerous and this unlocks powerful features such as `SharedArrayBuffer`,
+`performance.measureMemory` and the JS Self-Profiling API which could otherwise be
+used for Spectre-like attacks. It also prevents modifying `document.domain`.
+
+### Cross Origin Embedder Policy {: #coep }
+[Cross Origin Embedder
+Policy (COEP)](https://wicg.github.io/cross-origin-embedder-policy/) prevents a
+document from loading any cross-origin resources that don't explicitly grant
+the document permission (using CORP or CORS). With this feature, you can declare
+that a document cannot load such resources. 
+
+![How COEP works](coep.png)
+
+To activate this policy, append the following HTTP header to the document:
+
+```http
+Cross-Origin-Embedder-Policy: require-corp
+```
+
+The `require-corp` keyword is the only accepted value for COEP. This enforces
+the policy that the document can only load resources from the same origin, or
+resources explicitly marked as loadable from another origin.
+
+For resources to be loadable from another origin, they need to support either
+Cross Origin Resource Sharing (CORS) or Cross Origin Resource Policy (CORP).
+
+### Cross Origin Resource Sharing {: #cors }
+If a cross origin resource supports [Cross Origin Resource Sharing
+(CORS)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS), you may use the
+[`crossorigin`
+attribute](https://developer.mozilla.org/docs/Web/HTML/Attributes/crossorigin)
+to load it to your web page without being blocked by COEP.
+
+```html
+<img src="https://third-party.example.com/image.jpg" crossorigin>
+```
+
+For example, if this image resource is served with CORS headers, use the
+`crossorigin` attribute so that the request to fetch the resource will use [CORS
+mode](https://developer.mozilla.org/docs/Web/API/Request/mode). This also
+prevents the image from being loaded unless it sets CORS headers.
+
+Similarly, you may fetch cross origin data through the `fetch()` method, which
+doesn't require special handling as long as the server responds with [the right
+HTTP
+headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#The_HTTP_response_headers).
+
+### Cross Origin Resource Policy {: #corp }
+[Cross Origin Resource Policy
+(CORP)](https://developer.mozilla.org/docs/Web/HTTP/Cross-Origin_Resource_Policy_%28CORP%29)
+was originally introduced as an opt-in to protect your resources from being
+loaded by another origin. In the context of COEP, CORP can specify the resource
+owner's policy for who can load a resource.
+
+The `Cross-Origin-Resource-Policy` header takes three possible values:
+
+```http
+Cross-Origin-Resource-Policy: same-site
+Cross-Origin-Resource-Policy: same-origin
+Cross-Origin-Resource-Policy: cross-origin
+```
```

Comment:
Please remove this. It's redundant and it's not a code sample.

Reply:
I left these intentionally because i thought some developers won't understand what to do with the http header without actual sample. If you have other alternative to solve that, I'd happy to use the idea.

Reply:
The `Cross-Origin-Resource-Policy` header takes three possible values:

    ```http
    Cross-Origin-Resource-Policy: same-site
    ```

    `same-site`: Resources can only be loaded from the same site.

    ```http
    Cross-Origin-Resource-Policy: same-origin
    ```

    `same-origin`: Resources can only be loaded from the same origin.

    ```http
    Cross-Origin-Resource-Policy: cross-origin
    ```

    `cross-origin`: Resources can be loaded by any website. ([This
    value](https://mikewest.github.io/corpp/#integration-fetch) was added to the
    CORP spec along with COEP.)

Reply:
Thanks for the suggestion. I like it. Fixed.